railties/lib/rails/test_unit/minitest_plugin.rb

```diff
@@ -1,14 +1,50 @@
-require "minitest"
+require "active_support/core_ext/module/attribute_accessors"
 require "rails/test_unit/reporter"
+require "rails/test_unit/test_requirer"
 
-def Minitest.plugin_rails_init(options)
-  self.reporter << Rails::TestUnitReporter.new(options[:io], options)
-  if $rails_test_runner && (method = $rails_test_runner.find_method)
-    options[:filter] = method
+module Minitest
+  def self.plugin_rails_options(opts, options)
+    opts.separator ""
+    opts.separator "Usage: bin/rails test [options] [files or directories]"
+    opts.separator "You can run a single test by appending a line number to a filename:"
+    opts.separator ""
+    opts.separator "    bin/rails test test/models/user_test.rb:27"
+    opts.separator ""
+    opts.separator "You can run multiple files and directories at the same time:"
+    opts.separator ""
+    opts.separator "    bin/rails test test/controllers test/integration/login_test.rb"
+    opts.separator ""
+
+    opts.separator "Rails options:"
+    opts.on("-e", "--environment [ENV]",
+            "Run tests in the ENV environment") do |env|
+      options[:environment] = env.strip
+    end
+
+    opts.on("-b", "--backtrace",
+            "Show the complete backtrace") do
+      options[:full_backtrace] = true
+    end
+
+    options[:patterns] = opts.order!
   end
 
-  if !($rails_test_runner && $rails_test_runner.show_backtrace?)
-    Minitest.backtrace_filter = Rails.backtrace_cleaner
+  def self.plugin_rails_init(options)
+    self.run_with_rails_extension = true
+
+    ENV["RAILS_ENV"] = options[:environment] || "test"
+
+    Rails::TestRequirer.require_files options[:patterns] unless run_with_autorun
+
+    unless options[:full_backtrace] || ENV["BACKTRACE"]
+      Minitest.backtrace_filter = Rails.backtrace_cleaner if Rails.respond_to?(:backtrace_cleaner)
```

Comment:
as you said the `respond_to?` is not very clear. Let's add a comment to describe why it's there.